meerk40t/balormk/livelightjob.py

```diff
@@ -325,13 +327,24 @@ def _light_elements(self, con, elements):
             return self._crosshairs(con)
 
         rotate = self._redlight_adjust_matrix()
+        path_length = 0.0
         for node in elems:
             if self.stopped:
                 return False
             if self.changed:
                 return True
             geometry = Geomstr(node.as_geometry())
 
+            if self.mode == "full":
+                # Is path length too large for full?
+                for segment in range(0, len(geometry)):
+                    segment_length = geometry.length(segment)
```

Comment:
This `length()` call can be expensive. Like prohibitively, especially if you have cubic curves since they lack a closed form solution. You probably want to make this metric called based on `len(geometry)` or so. Also, if it's huge calculating the bounds might be somewhat costly too.

Reply:
The original desire for this pr was because in the 0.8.x branch it would freeze during live full.  In the 0.9.x branch, it will run slowly on large paths but not freeze up the program.  The user can see this and switch modes manually so I don't think choosing that for them is useful.  Closing this pr.